dev/io.openliberty.jakarta.data.1.0/src/jakarta/data/repository/Pagination.java

```diff
@@ -14,104 +14,55 @@
 
 import java.util.Collections;
 import java.util.List;
-import java.util.Objects;
 import java.util.stream.Collectors;
 import java.util.stream.StreamSupport;
 
+import jakarta.data.repository.Pageable.Cursor;
+import jakarta.data.repository.Pageable.Mode;
+
 /**
  * Method signatures are copied from jakarta.data.repository.Pageable from the Jakarta Data repo.
  */
-class Pagination implements Pageable {
-    private final Cursor cursor;
-    private final Mode mode;
-    private final List<Sort> order;
-    private final long pageNumber;
-    private final int pageSize;
-    private final int hash;
-
-    Pagination(long pageNumber, int pageSize, List<Sort> order, Mode mode, Cursor cursor) {
-        if (pageNumber < 1)
-            throw new IllegalArgumentException("pageNumber: " + pageNumber);
-        if (pageSize < 1)
-            throw new IllegalArgumentException("pageSize: " + pageSize);
+record Pagination(long page,
+                int size,
+                List<Sort> sorts,
+                Mode mode,
+                Cursor cursor)
+                implements Pageable {
+
+    Pagination {
+        if (page < 1)
+            throw new IllegalArgumentException("pageNumber: " + page);
```

Comment:
Since the `pageNumber` parameter was renamed to `page`, I would suggest changing this line to:
```
throw new IllegalArgumentException("page: " + page);
```
to be consistent.

Reply:
The message needs to match the Jakarta Data beta 3 driver. Possibly it should change there, but we need to remain consistent with it until it does.